### DISCUSSION TITLE
华为手机升级到4.2.0.129后无法使用

### DISCUSSION THREAD

**Discussion by saiapple** [2dust/v2rayNG#3202] [General]
在提出问题前请先自行排除服务器端问题，同时也请通过搜索确认是否有人提出过相同问题。

### 预期行为
描述你认为应该发生什么

### 实际行为
客户端启动服务失败

### 复现方法
华为mate50升级到4.2.0.129

### 日志信息
--------- beginning of crash
06-01 08:07:12.144 E/AndroidRuntime(17963): FATAL EXCEPTION: main
06-01 08:07:12.144 E/AndroidRuntime(17963): Process: com.v2ray.ang:RunSoLibV2RayDaemon, PID: 17963
06-01 08:07:12.144 E/AndroidRuntime(17963): android.app.ForegroundServiceDidNotStartInTimeException: Context.startForegroundService() did not then call Service.startForeground(): ServiceRecord{d485a48 u0 com.v2ray.ang/.service.V2RayVpnService}
06-01 08:07:12.144 E/AndroidRuntime(17963): 	at android.app.ActivityThread.throwRemoteServiceException(ActivityThread.java:2596)
06-01 08:07:12.144 E/AndroidRuntime(17963): 	at android.app.ActivityThread.access$4000(ActivityThread.java:313)
06-01 08:07:12.144 E/AndroidRuntime(17963): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:3013)
06-01 08:07:12.144 E/AndroidRuntime(17963): 	at android.os.Handler.dispatchMessage(Handler.java:117)
06-01 08:07:12.144 E/AndroidRuntime(17963): 	at android.os.Looper.loopOnce(Looper.java:205)
06-01 08:07:12.144 E/AndroidRuntime(17963): 	at android.os.Looper.loop(Looper.java:293)
06-01 08:07:12.144 E/AndroidRuntime(17963): 	at android.app.ActivityThread.loopProcess(ActivityThread.java:9986)
06-01 08:07:12.144 E/AndroidRuntime(17963): 	at android.app.ActivityThread.main(ActivityThread.java:9975)
06-01 08:07:12.144 E/AndroidRuntime(17963): 	at java.lang.reflect.Method.invoke(Native Method)
06-01 08:07:12.144 E/AndroidRuntime(17963): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:586)
06-01 08:07:12.144 E/AndroidRuntime(17963): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1240)
06-01 08:10:14.867 E/AndroidRuntime(21617): FATAL EXCEPTION: main
06-01 08:10:14.867 E/AndroidRuntime(21617): Process: com.v2ray.ang:RunSoLibV2RayDaemon, PID: 21617
06-01 08:10:14.867 E/AndroidRuntime(21617): android.app.ForegroundServiceDidNotStartInTimeException: Context.startForegroundService() did not then call Service.startForeground(): ServiceRecord{47f4bc6 u0 com.v2ray.ang/.service.V2RayVpnService}
06-01 08:10:14.867 E/AndroidRuntime(21617): 	at android.app.ActivityThread.throwRemoteServiceException(ActivityThread.java:2596)
06-01 08:10:14.867 E/AndroidRuntime(21617): 	at android.app.ActivityThread.access$4000(ActivityThread.java:313)
06-01 08:10:14.867 E/AndroidRuntime(21617): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:3013)
06-01 08:10:14.867 E/AndroidRuntime(21617): 	at android.os.Handler.dispatchMessage(Handler.java:117)
06-01 08:10:14.867 E/AndroidRuntime(21617): 	at android.os.Looper.loopOnce(Looper.java:205)
06-01 08:10:14.867 E/AndroidRuntime(21617): 	at android.os.Looper.loop(Looper.java:293)
06-01 08:10:14.867 E/AndroidRuntime(21617): 	at android.app.ActivityThread.loopProcess(ActivityThread.java:9986)
06-01 08:10:14.867 E/AndroidRuntime(21617): 	at android.app.ActivityThread.main(ActivityThread.java:9975)
06-01 08:10:14.867 E/AndroidRuntime(21617): 	at java.lang.reflect.Method.invoke(Native Method)
06-01 08:10:14.867 E/AndroidRuntime(21617): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:586)
06-01 08:10:14.867 E/AndroidRuntime(21617): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1240)
06-01 08:13:35.529 E/AndroidRuntime(26955): FATAL EXCEPTION: main
06-01 08:13:35.529 E/AndroidRuntime(26955): Process: com.v2ray.ang:RunSoLibV2RayDaemon, PID: 26955
06-01 08:13:35.529 E/AndroidRuntime(26955): android.app.ForegroundServiceDidNotStartInTimeException: Context.startForegroundService() did not then call Service.startForeground(): ServiceRecord{c26d6e1 u0 com.v2ray.ang/.service.V2RayVpnService}
06-01 08:13:35.529 E/AndroidRuntime(26955): 	at android.app.ActivityThread.throwRemoteServiceException(ActivityThread.java:2596)
06-01 08:13:35.529 E/AndroidRuntime(26955): 	at android.app.ActivityThread.access$4000(ActivityThread.java:313)
06-01 08:13:35.529 E/AndroidRuntime(26955): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:3013)
06-01 08:13:35.529 E/AndroidRuntime(26955): 	at android.os.Handler.dispatchMessage(Handler.java:117)
06-01 08:13:35.529 E/AndroidRuntime(26955): 	at android.os.Looper.loopOnce(Looper.java:205)
06-01 08:13:35.529 E/AndroidRuntime(26955): 	at android.os.Looper.loop(Looper.java:293)
06-01 08:13:35.529 E/AndroidRuntime(26955): 	at android.app.ActivityThread.loopProcess(ActivityThread.java:9986)
06-01 08:13:35.529 E/AndroidRuntime(26955): 	at android.app.ActivityThread.main(ActivityThread.java:9975)
06-01 08:13:35.529 E/AndroidRuntime(26955): 	at java.lang.reflect.Method.invoke(Native Method)
06-01 08:13:35.529 E/AndroidRuntime(26955): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:586)
06-01 08:13:35.529 E/AndroidRuntime(26955): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1240)
06-01 08:22:07.125 E/AndroidRuntime(10161): FATAL EXCEPTION: main
06-01 08:22:07.125 E/AndroidRuntime(10161): Process: com.v2ray.ang:RunSoLibV2RayDaemon, PID: 10161
06-01 08:22:07.125 E/AndroidRuntime(10161): android.app.ForegroundServiceDidNotStartInTimeException: Context.startForegroundService() did not then call Service.startForeground(): ServiceRecord{a6a7a5f u0 com.v2ray.ang/.service.V2RayVpnService}
06-01 08:22:07.125 E/AndroidRuntime(10161): 	at android.app.ActivityThread.throwRemoteServiceException(ActivityThread.java:2596)
06-01 08:22:07.125 E/AndroidRuntime(10161): 	at android.app.ActivityThread.access$4000(ActivityThread.java:313)
06-01 08:22:07.125 E/AndroidRuntime(10161): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:3013)
06-01 08:22:07.125 E/AndroidRuntime(10161): 	at android.os.Handler.dispatchMessage(Handler.java:117)
06-01 08:22:07.125 E/AndroidRuntime(10161): 	at android.os.Looper.loopOnce(Looper.java:205)
06-01 08:22:07.125 E/AndroidRuntime(10161): 	at android.os.Looper.loop(Looper.java:293)
06-01 08:22:07.125 E/AndroidRuntime(10161): 	at android.app.ActivityThread.loopProcess(ActivityThread.java:9986)
06-01 08:22:07.125 E/AndroidRuntime(10161): 	at android.app.ActivityThread.main(ActivityThread.java:9975)
06-01 08:22:07.125 E/AndroidRuntime(10161): 	at java.lang.reflect.Method.invoke(Native Method)
06-01 08:22:07.125 E/AndroidRuntime(10161): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:586)
06-01 08:22:07.125 E/AndroidRuntime(10161): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1240)
--------- beginning of main
06-01 08:22:32.009 E/com.v2ray.ang(17935): Not starting debugger since process cannot load the jdwp agent.

通过`adb logcat -s com.v2ray.ang GoLog V2rayConfigUtilGoLog Main`获取日志。请自行删减日志中可能出现的敏感信息。

如果问题可重现，建议先执行`adb logcat -c`清空系统日志再执行上述命令，再操作重现问题。
```
在这里粘贴日志
```
</details>

**Comment by freewinding** [comment]
鸿蒙的安全度较高，我分析https都一样不被信任
